### ISSUE TITLE
We cannot register any user since KC 22.0.0 update

### ISSUE THREAD

**Issue by viyullas** [RedFroggy/keycloak-verify-email-by-code#14]
After filling the registration form the email code is send but the form to enter it is not presented, instead we get an error 500

**Comment by taaem** [comment]
+1 I'm also having this issue according to the logs:
```
[org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-17) Uncaught server error: java.lang.NoSuchMethodError: 'javax.ws.rs.core.Response org.keycloak.forms.login.LoginFormsProvider.createForm(java.lang.String)'
2023-07-14T12:11:38.615981885Z 	at fr.redfroggy.keycloak.requiredactions.VerifyEmailByCode.createFormChallenge(VerifyEmailByCode.java:130)
2023-07-14T12:11:38.615988969Z 	at fr.redfroggy.keycloak.requiredactions.VerifyEmailByCode.sendVerifyEmailAndCreateForm(VerifyEmailByCode.java:192)
2023-07-14T12:11:38.615993434Z 	at fr.redfroggy.keycloak.requiredactions.VerifyEmailByCode.requiredActionChallenge(VerifyEmailByCode.java:96)
2023-07-14T12:11:38.615998025Z 	at org.keycloak.services.managers.AuthenticationManager.executeAction(AuthenticationManager.java:1334)
2023-07-14T12:11:38.616002177Z 	at org.keycloak.services.managers.AuthenticationManager.lambda$executionActions$18(AuthenticationManager.java:1281)
2023-07-14T12:11:38.616006337Z 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
2023-07-14T12:11:38.616010327Z 	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:400)
2023-07-14T12:11:38.616014226Z 	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
2023-07-14T12:11:38.616018209Z 	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
2023-07-14T12:11:38.616022087Z 	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
2023-07-14T12:11:38.616026028Z 	at java.base/java.util.stream.DistinctOps$1$2.end(DistinctOps.java:168)
2023-07-14T12:11:38.616030041Z 	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
2023-07-14T12:11:38.616033990Z 	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:528)
2023-07-14T12:11:38.616110599Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
2023-07-14T12:11:38.616120858Z 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
2023-07-14T12:11:38.616125292Z 	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
2023-07-14T12:11:38.616129248Z 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
2023-07-14T12:11:38.616133136Z 	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
2023-07-14T12:11:38.616137148Z 	at org.keycloak.services.managers.AuthenticationManager.executionActions(AuthenticationManager.java:1282)
2023-07-14T12:11:38.616142923Z 	at org.keycloak.services.managers.AuthenticationManager.actionRequired(AuthenticationManager.java:1170)
2023-07-14T12:11:38.616147099Z 	at org.keycloak.services.managers.AuthenticationManager.nextActionAfterAuthentication(AuthenticationManager.java:1017)
2023-07-14T12:11:38.616151098Z 	at org.keycloak.services.resources.LoginActionsService.processRequireAction(LoginActionsService.java:1069)
2023-07-14T12:11:38.616155213Z 	at org.keycloak.services.resources.LoginActionsService.requiredActionGET(LoginActionsService.java:1051)
2023-07-14T12:11:38.616159251Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-07-14T12:11:38.616163208Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-07-14T12:11:38.616167219Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-07-14T12:11:38.616171302Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2023-07-14T12:11:38.616175305Z 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:154)
2023-07-14T12:11:38.616179295Z 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:118)
2023-07-14T12:11:38.616183236Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:560)
2023-07-14T12:11:38.616187254Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:452)
2023-07-14T12:11:38.616191246Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:413)
2023-07-14T12:11:38.616195264Z 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
2023-07-14T12:11:38.616199350Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:415)
2023-07-14T12:11:38.616203445Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:378)
2023-07-14T12:11:38.616207535Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:174)
2023-07-14T12:11:38.616211555Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:131)
2023-07-14T12:11:38.616224601Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:33)
2023-07-14T12:11:38.616229180Z 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:429)
2023-07-14T12:11:38.616233215Z 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:240)
2023-07-14T12:11:38.616237778Z 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:154)
2023-07-14T12:11:38.616241861Z 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
2023-07-14T12:11:38.616245933Z 	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
2023-07-14T12:11:38.616250028Z 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:229)
2023-07-14T12:11:38.616253958Z 	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:82)
2023-07-14T12:11:38.616257963Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:147)
2023-07-14T12:11:38.616262049Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:84)
2023-07-14T12:11:38.616266144Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:44)
2023-07-14T12:11:38.616270279Z 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
2023-07-14T12:11:38.616275607Z 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2023-07-14T12:11:38.616279764Z 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
2023-07-14T12:11:38.616283848Z 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:58)
2023-07-14T12:11:38.616287949Z 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:36)
2023-07-14T12:11:38.616291990Z 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
2023-07-14T12:11:38.616295919Z 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
2023-07-14T12:11:38.616299924Z 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
2023-07-14T12:11:38.616303954Z 	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$0(QuarkusRequestFilter.java:82)
2023-07-14T12:11:38.616307997Z 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
2023-07-14T12:11:38.616312028Z 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
2023-07-14T12:11:38.616316198Z 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
2023-07-14T12:11:38.616320233Z 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
2023-07-14T12:11:38.616328747Z 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
2023-07-14T12:11:38.616333757Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2023-07-14T12:11:38.616339444Z 	at java.base/java.lang.Thread.run(Thread.java:833)
```
This seems due to Keycloak migrating from JavaEE to JakartaEE in version 22 (https://www.keycloak.org/docs/latest/release_notes/index.html#transition-from-java-ee-to-jakarta-ee) and therefore the method type is `jakarta.ws.rs.core.Response createForm(String form)` now.

**Comment by jvjimenez** [comment]
> +1 I'm also having this issue according to the logs:
> 
> ```
> [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-17) Uncaught server error: java.lang.NoSuchMethodError: 'javax.ws.rs.core.Response org.keycloak.forms.login.LoginFormsProvider.createForm(java.lang.String)'
> 2023-07-14T12:11:38.615981885Z 	at fr.redfroggy.keycloak.requiredactions.VerifyEmailByCode.createFormChallenge(VerifyEmailByCode.java:130)
> 2023-07-14T12:11:38.615988969Z 	at fr.redfroggy.keycloak.requiredactions.VerifyEmailByCode.sendVerifyEmailAndCreateForm(VerifyEmailByCode.java:192)
> 2023-07-14T12:11:38.615993434Z 	at fr.redfroggy.keycloak.requiredactions.VerifyEmailByCode.requiredActionChallenge(VerifyEmailByCode.java:96)
> 2023-07-14T12:11:38.615998025Z 	at org.keycloak.services.managers.AuthenticationManager.executeAction(AuthenticationManager.java:1334)
> 2023-07-14T12:11:38.616002177Z 	at org.keycloak.services.managers.AuthenticationManager.lambda$executionActions$18(AuthenticationManager.java:1281)
> 2023-07-14T12:11:38.616006337Z 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
> 2023-07-14T12:11:38.616010327Z 	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:400)
> 2023-07-14T12:11:38.616014226Z 	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
> 2023-07-14T12:11:38.616018209Z 	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
> 2023-07-14T12:11:38.616022087Z 	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
> 2023-07-14T12:11:38.616026028Z 	at java.base/java.util.stream.DistinctOps$1$2.end(DistinctOps.java:168)
> 2023-07-14T12:11:38.616030041Z 	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
> 2023-07-14T12:11:38.616033990Z 	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:528)
> 2023-07-14T12:11:38.616110599Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
> 2023-07-14T12:11:38.616120858Z 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
> 2023-07-14T12:11:38.616125292Z 	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
> 2023-07-14T12:11:38.616129248Z 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
> 2023-07-14T12:11:38.616133136Z 	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
> 2023-07-14T12:11:38.616137148Z 	at org.keycloak.services.managers.AuthenticationManager.executionActions(AuthenticationManager.java:1282)
> 2023-07-14T12:11:38.616142923Z 	at org.keycloak.services.managers.AuthenticationManager.actionRequired(AuthenticationManager.java:1170)
> 2023-07-14T12:11:38.616147099Z 	at org.keycloak.services.managers.AuthenticationManager.nextActionAfterAuthentication(AuthenticationManager.java:1017)
> 2023-07-14T12:11:38.616151098Z 	at org.keycloak.services.resources.LoginActionsService.processRequireAction(LoginActionsService.java:1069)
> 2023-07-14T12:11:38.616155213Z 	at org.keycloak.services.resources.LoginActionsService.requiredActionGET(LoginActionsService.java:1051)
> 2023-07-14T12:11:38.616159251Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 2023-07-14T12:11:38.616163208Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
> 2023-07-14T12:11:38.616167219Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 2023-07-14T12:11:38.616171302Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
> 2023-07-14T12:11:38.616175305Z 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:154)
> 2023-07-14T12:11:38.616179295Z 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:118)
> 2023-07-14T12:11:38.616183236Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:560)
> 2023-07-14T12:11:38.616187254Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:452)
> 2023-07-14T12:11:38.616191246Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:413)
> 2023-07-14T12:11:38.616195264Z 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
> 2023-07-14T12:11:38.616199350Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:415)
> 2023-07-14T12:11:38.616203445Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:378)
> 2023-07-14T12:11:38.616207535Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:174)
> 2023-07-14T12:11:38.616211555Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:131)
> 2023-07-14T12:11:38.616224601Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:33)
> 2023-07-14T12:11:38.616229180Z 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:429)
> 2023-07-14T12:11:38.616233215Z 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:240)
> 2023-07-14T12:11:38.616237778Z 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:154)
> 2023-07-14T12:11:38.616241861Z 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
> 2023-07-14T12:11:38.616245933Z 	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
> 2023-07-14T12:11:38.616250028Z 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:229)
> 2023-07-14T12:11:38.616253958Z 	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:82)
> 2023-07-14T12:11:38.616257963Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:147)
> 2023-07-14T12:11:38.616262049Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:84)
> 2023-07-14T12:11:38.616266144Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:44)
> 2023-07-14T12:11:38.616270279Z 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
> 2023-07-14T12:11:38.616275607Z 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
> 2023-07-14T12:11:38.616279764Z 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
> 2023-07-14T12:11:38.616283848Z 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:58)
> 2023-07-14T12:11:38.616287949Z 	at io.quarkus.vertx.http.runtime.options.HttpServerCommonHandlers$1.handle(HttpServerCommonHandlers.java:36)
> 2023-07-14T12:11:38.616291990Z 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
> 2023-07-14T12:11:38.616295919Z 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
> 2023-07-14T12:11:38.616299924Z 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
> 2023-07-14T12:11:38.616303954Z 	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$0(QuarkusRequestFilter.java:82)
> 2023-07-14T12:11:38.616307997Z 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
> 2023-07-14T12:11:38.616312028Z 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
> 2023-07-14T12:11:38.616316198Z 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
> 2023-07-14T12:11:38.616320233Z 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
> 2023-07-14T12:11:38.616328747Z 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
> 2023-07-14T12:11:38.616333757Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
> 2023-07-14T12:11:38.616339444Z 	at java.base/java.lang.Thread.run(Thread.java:833)
> ```
> 
> This seems due to Keycloak migrating from JavaEE to JakartaEE in version 22 (https://www.keycloak.org/docs/latest/release_notes/index.html#transition-from-java-ee-to-jakarta-ee) and therefore the method type is `jakarta.ws.rs.core.Response createForm(String form)` now.

+1

**Comment by dajay** [comment]
Hello,
New release is available [22.0.0](https://github.com/RedFroggy/keycloak-verify-email-by-code/releases/tag/22.0.0).
Thanks
